wasm.md

```diff
@@ -1447,28 +1443,30 @@ A table index is optional and will be default to zero.
     syntax ElemDefn ::= "(" "elem"     TextFormatIdx Offset ElemSegment ")"
                       | "(" "elem"                   Offset ElemSegment ")"
                       |     "elem" "{" TextFormatIdx        ElemSegment "}"
-    syntax Stmt     ::= #initElements ( Int, Int, ElemSegment )
- // -----------------------------------------------------------
+    syntax Stmt     ::= #initElements ( Int, Int, Map, Map, ElemSegment )
+ // ---------------------------------------------------------------------
     // Default to table with index 0.
     rule <k> ( elem        OFFSET      ELEMSEGMENT ) =>     ( elem 0 OFFSET ELEMSEGMENT ) ... </k>
     rule <k> ( elem TABIDX IS:Instr    ELEMSEGMENT ) => IS ~> elem { TABIDX ELEMSEGMENT } ... </k>
     rule <k> ( elem TABIDX (offset IS) ELEMSEGMENT ) => IS ~> elem { TABIDX ELEMSEGMENT } ... </k>
 
-    rule <k> elem { TABIDX ELEMSEGMENT } => #initElements ( ADDR, OFFSET, ELEMSEGMENT ) ... </k>
+    rule <k> elem { TABIDX ELEMSEGMENT } => #initElements ( ADDR, OFFSET, FADDRS, FIDS, ELEMSEGMENT ) ... </k>
          <curModIdx> CUR </curModIdx>
          <valstack> < i32 > OFFSET : STACK => STACK </valstack>
          <moduleInst>
-           <modIdx> CUR </modIdx>
-           <tabIds> IDS </tabIds>
-           <tabIndices> #ContextLookup(IDS, TABIDX) |-> ADDR </tabIndices>
+           <modIdx> CUR  </modIdx>
+           <funcIds> FIDS </funcIds>
+           <funcIndices> FADDRS </funcIndices>
+           <tabIds>  TIDS </tabIds>
+           <tabIndices> #ContextLookup(TIDS, TABIDX) |-> ADDR </tabIndices>
            ...
          </moduleInst>
 
-    rule <k> #initElements ( ADDR, OFFSET, .ElemSegment ) => . ... </k>
-    rule <k> #initElements ( ADDR, OFFSET,  E ES        ) => #initElements ( ADDR, OFFSET +Int 1, ES ) ... </k>
+    rule <k> #initElements (    _,     _ ,      _,   _, .ElemSegment ) => . ... </k>
+    rule <k> #initElements ( ADDR, OFFSET, FADDRS, IDS,  E ES        ) => #initElements ( ADDR, OFFSET +Int 1, FADDRS, IDS, ES ) ... </k>
```

Comment:
Not sure as this function is in `<k>` cell, why do we need to pass `FADDRS` and `IDS` to the function, rather than directly look it up in the configuration?

Reply:
I figured that if we are using a helper `Stmt`, we might as well give it as little context as possible. (Basically, it just makes the rule smaller, as it is now acting just on the store and not the module it is in).